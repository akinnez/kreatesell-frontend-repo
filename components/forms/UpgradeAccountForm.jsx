import { Button } from "components/button/Button";
import Image from "next/image";
import {
	ActiveTick,
	InactiveMasterCard,
	InactivePaypal,
	ActiveStripe,
} from "utils";
import { RightArrow } from "utils/icons/RightArrow";

export const UpgradeAccountForm = () => {
	return (
		<>
			<div className="px-0 md:px-8">
				<div className="text-center mb-4">
					<h3 className="text-black-100 font-bold text-xl">
						Upgrade Your Account
					</h3>
					<h4 className="text-black-100 pt-2">BUSINESS</h4>

					<div className="divider"></div>

					<div className="text-base-green-200 font-bold text-2xl">
						<sup className="font-normal text-xs text-black-100">NGN</sup> 4,167
						<sub className="font-normal text-xs text-black-100">/ Month</sub>
					</div>
				</div>

				<form className="px-2 md:px-8 pt-4">
					<div className="text-primary-blue font-medium text-lg">
						Payment Details
					</div>
					<div className="divider"></div>

					<div>
						<div>Select Currency</div>
						<p className="text-base-gray-200 text-xs pt-2 md:pt-0 md:text-sm">
							Select your preferred currency and get price equivalent
						</p>
					</div>

					<div className="grid gap-4 grid-cols-3 md:grid-cols-6 pt-3">
						<div className="activeCard px-2 flex items-center justify-between">
							<p className="pt-2">NGN</p>
							<div>
								<Image src={ActiveTick} alt="active" width="16" height="16" />
							</div>
						</div>
						<div className="card p-2 flex items-center">USD</div>
						<div className="card p-2 flex items-center">GBP</div>
						<div className="card p-2 flex items-center">KES</div>
						<div className="card p-2 flex items-center">ZAR</div>
						<div className="card p-2 flex items-center">GHS</div>
					</div>

					<div className="pt-6">
						<div>Payment Method</div>
						<p className="text-base-gray-200 text-xs pt-2 md:pt-0 md:text-sm">
							Select your preferred payment method
						</p>
					</div>

					<div className="grid gap-4 grid-cols-3 pt-3">
						<div className="activeCard p-2 flex justify-between items-center">
							<div>
								<Image src={ActiveStripe} alt="stripe" />
							</div>
							<div>
								<Image src={ActiveTick} alt="active" width="16" height="16" />
							</div>
						</div>

						<div className="flex justify-center card p-2">
							<Image src={InactivePaypal} alt="stripe" />
						</div>

						<div className="flex justify-center card p-2">
							<Image src={InactiveMasterCard} alt="stripe" />
						</div>
					</div>

					<div className="priceMenu my-6 py-3 px-8">
						<div className="flex justify-between pt-2">
							<p>SubTotal</p>
							<p>NGN 4,167</p>
						</div>
						<div className="divider"> </div>
						<div className="flex justify-between">
							<p>Total</p>
							<p className="text-primary-blue font-medium">NGN 4,167</p>
						</div>
					</div>

					<div className="w-full">
						<Button
							text="Pay NGN 4,167"
							bgColor="blue"
							style={{ width: "100%" }}
							icon={<RightArrow />}
						/>
					</div>
				</form>
			</div>

			<style jsx>{`
				.activeCard {
					border: 1px solid #2dc071;
					border-radius: 0.5rem;
					cursor: pointer;
					color: #8c8c8c;
					font-size: 12px;
				}

				.card {
					border-radius: 0.5rem;
					border: 1px solid #f0f0f0;
					cursor: pointer;
					color: #8c8c8c;
					font-size: 12px;
				}

				.priceMenu {
					box-shadow: 0px 20px 200px rgba(34, 34, 34, 0.1);
					background: #ffffff;
					color: #262626;
				}
			`}</style>
		</>
	);
};
