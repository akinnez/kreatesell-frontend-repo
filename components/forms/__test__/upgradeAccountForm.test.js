import { UpgradeAccountForm } from "../UpgradeAccountForm";
import renderer from "react-test-renderer";

it("renders a snapshot of <UpgradeAccountForm />", () => {
  const form = renderer.create(<UpgradeAccountForm />).toJSON();
  expect(form).toMatchSnapshot();
});
