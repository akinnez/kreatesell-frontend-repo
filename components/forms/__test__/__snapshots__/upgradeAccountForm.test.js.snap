// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <UpgradeAccountForm /> 1`] = `
<div
  className="jsx-882199799 px-0 md:px-8"
>
  <div
    className="jsx-882199799 text-center mb-4"
  >
    <h3
      className="jsx-882199799 text-black-100 font-bold text-xl"
    >
      Upgrade Your Account
    </h3>
    <h4
      className="jsx-882199799 text-black-100 pt-2"
    >
      BUSINESS
    </h4>
    <div
      className="jsx-882199799 divider"
    />
    <div
      className="jsx-882199799 text-base-green-200 font-bold text-2xl"
    >
      <sup
        className="jsx-882199799 font-normal text-xs text-black-100"
      >
        NGN
      </sup>
       4,167
      <sub
        className="jsx-882199799 font-normal text-xs text-black-100"
      >
        / Month
      </sub>
    </div>
  </div>
  <form
    className="jsx-882199799 px-2 md:px-8 pt-4"
  >
    <div
      className="jsx-882199799 text-primary-blue font-medium text-lg"
    >
      Payment Details
    </div>
    <div
      className="jsx-882199799 divider"
    />
    <div
      className="jsx-882199799"
    >
      <div
        className="jsx-882199799"
      >
        Select Currency
      </div>
      <p
        className="jsx-882199799 text-base-gray-200 text-xs pt-2 md:pt-0 md:text-sm"
      >
        Select your preferred currency and get price equivalent
      </p>
    </div>
    <div
      className="jsx-882199799 grid gap-4 grid-cols-3 md:grid-cols-6 pt-3"
    >
      <div
        className="jsx-882199799 activeCard px-2 flex items-center justify-between"
      >
        <p
          className="jsx-882199799 pt-2"
        >
          NGN
        </p>
        <div
          className="jsx-882199799"
        >
          <img
            src="[jest.find('../utils')]"
          />
        </div>
      </div>
      <div
        className="jsx-882199799 card p-2 flex items-center"
      >
        USD
      </div>
      <div
        className="jsx-882199799 card p-2 flex items-center"
      >
        GBP
      </div>
      <div
        className="jsx-882199799 card p-2 flex items-center"
      >
        KES
      </div>
      <div
        className="jsx-882199799 card p-2 flex items-center"
      >
        ZAR
      </div>
      <div
        className="jsx-882199799 card p-2 flex items-center"
      >
        GHS
      </div>
    </div>
    <div
      className="jsx-882199799 pt-6"
    >
      <div
        className="jsx-882199799"
      >
        Payment Method
      </div>
      <p
        className="jsx-882199799 text-base-gray-200 text-xs pt-2 md:pt-0 md:text-sm"
      >
        Select your preferred payment method
      </p>
    </div>
    <div
      className="jsx-882199799 grid gap-4 grid-cols-3 pt-3"
    >
      <div
        className="jsx-882199799 activeCard p-2 flex justify-between items-center"
      >
        <div
          className="jsx-882199799"
        >
          <img
            src="[jest.find('../utils')]"
          />
        </div>
        <div
          className="jsx-882199799"
        >
          <img
            src="[jest.find('../utils')]"
          />
        </div>
      </div>
      <div
        className="jsx-882199799 flex justify-center card p-2"
      >
        <img
          src="[jest.find('../utils')]"
        />
      </div>
      <div
        className="jsx-882199799 flex justify-center card p-2"
      >
        <img
          src="[jest.find('../utils')]"
        />
      </div>
    </div>
    <div
      className="jsx-882199799 priceMenu my-6 py-3 px-8"
    >
      <div
        className="jsx-882199799 flex justify-between pt-2"
      >
        <p
          className="jsx-882199799"
        >
          SubTotal
        </p>
        <p
          className="jsx-882199799"
        >
          NGN 4,167
        </p>
      </div>
      <div
        className="jsx-882199799 divider"
      >
         
      </div>
      <div
        className="jsx-882199799 flex justify-between"
      >
        <p
          className="jsx-882199799"
        >
          Total
        </p>
        <p
          className="jsx-882199799 text-primary-blue font-medium"
        >
          NGN 4,167
        </p>
      </div>
    </div>
    <div
      className="jsx-882199799 w-full"
    >
      <button
        className="
			undefined
			button
			false 
			btnBlue
			false
			"
        style={
          Object {
            "width": "100%",
          }
        }
        type="submit"
      >
        <div
          className="buttonValue"
        >
          Pay NGN 4,167
          <div
            className="buttonIcon"
          >
            <svg
              fill="none"
              height="10"
              viewBox="0 0 13 10"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M0.97144 4.45143C0.705203 4.48809 0.5 4.71969 0.5 4.99993C0.5 5.30564 0.744208 5.55347 1.04545 5.55347L10.6333 5.55347L7.16979 9.05423L7.11685 9.1162C6.95797 9.33257 6.97495 9.64013 7.1682 9.83706C7.38077 10.0537 7.72613 10.0544 7.93958 9.83868L12.3317 5.3998C12.3586 5.37364 12.3829 5.34471 12.4041 5.31346C12.5509 5.09748 12.5294 4.7992 12.3396 4.60737L7.93955 0.161287L7.87826 0.107824C7.66438 -0.0524988 7.36139 -0.0339699 7.16816 0.162973C6.95561 0.379612 6.95635 0.730097 7.16982 0.945805L10.6341 4.44638L1.04545 4.44638L0.97144 4.45143Z"
                fill="white"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
  </form>
</div>
`;
